File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res=[1 for x in range(len(nums))]
        for x in range(1,len(res)):
            res[x]=res[x-1]*nums[x-1]
        temp=1
        for x in range(len(res)-1,-1,-1):
            res[x]*=temp
            temp*=nums[x]
        return res



        #1,2,4,6
        #1,1,2,8
        #48,24,12,8

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer,Integer> map = new HashMap<>();
        int prefix=0;
        int res=0;
        map.put(0,1);
        for(int i:nums){
            prefix+=i;
            int complement=prefix-k;
            res+=map.getOrDefault(complement,0);
            map.put(prefix,map.getOrDefault(prefix,0)+1);

        }
        return res;
    }
}